src/harness/fake_db_server.py

```diff
@@ -0,0 +1,78 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""A fake implementation of database.
+
+A local test server could be run by using "python fake_db_server.py".
+
+"""
+
+import json
+import threading
+from BaseHTTPServer import HTTPServer, BaseHTTPRequestHandler
+
+class FakeDatabaseTestHarness(threading.Thread):
+
+  def __init__(self, url = None, database_file = None):
+    """Test Harness acting as a Http Server to receive Pull/GET
+    Requests from SAS Under Test
+    parameters: is the dict variable for database.
+    schema of parameters: {
+    "path": path to the database file
```

Comment:
Will we always know the path in advance? (I can't think of any obvious exceptions, but I'm not an expert in this area).

Will there always be only one file? I think for the ULS database, there are multiple files that need to be combined -- but again, this is not my area of expertise.

You may need a different DB class for each type of DB although I'm not sure at this time what the best approach is.

Reply:
Yes,   the path is known as the testcase creates exclusion zone database          Yes, single file is created for fake DB implementation          Different types of data are dumped into differrent directory in fake   database so no need to create DB class. For example:   fakedatabase/exclusionzone and fakedatabase/ppa